zebra/redistribute.c

```diff
@@ -617,17 +662,25 @@ void zebra_interface_vrf_update_add(struct interface *ifp, vrf_id_t old_vrf_id)
 {
 	struct listnode *node, *nnode;
 	struct zserv *client;
+	char buf[256];
 
 	if (IS_ZEBRA_DEBUG_EVENT)
-		zlog_debug(
-			"MESSAGE: ZEBRA_INTERFACE_VRF_UPDATE/ADD %s VRF Id %u -> %u",
-			ifp->name, old_vrf_id, ifp->vrf_id);
+		snprintf(buf, sizeof(buf),
+			 "ZEBRA_INTERFACE_VRF_UPDATE/ADD %s VRF Id %u -> %u",
+			 ifp->name, old_vrf_id, ifp->vrf_id);
 
 	for (ALL_LIST_ELEMENTS(zrouter.client_list, node, nnode, client)) {
 		/* Need to add if the client is interested in the new VRF. */
+		if (!vrf_bitmap_check(client->vrf_reg_info, ifp->vrf_id))
+			continue;
+
+		if (IS_ZEBRA_DEBUG_EVENT)
+			zlog_debug("%s to zclient: %s", buf,
+			    zebra_route_string(client->proto));
+
 		client->ifadd_cnt++;
 		zsend_interface_add(client, ifp);
-		zsend_interface_addresses(client, ifp);
+		zsend_interface_link_params(client, ifp);
```

Comment:
Shouldn't we send both the interface addresses and link params here? In any case, any change here probably deserves its own commit.

Reply:
Hi @karamalla0406 ,
with that change: 'zsend_interface_link_params()', I understand you send now more information along with the interface main inforamtion.
now my question is: do you intend to give more information on the interface to the zclients in the same message ? I mean for instance vxlan and gre interface information should transmitted ? Or do you think some specific message should be registered ?

Reply:
> Shouldn't we send both the interface addresses and link params here? In any case, any change here probably deserves its own commit.

The interface address is sent in 

> Shouldn't we send both the interface addresses and link params here? In any case, any change here probably deserves its own commit.

Interface addresses are sent in zebra_interface_address_add_update. It will be a duplicate, so I removed it in here.

Reply:
> Hi @karamalla0406 ,
> with that change: 'zsend_interface_link_params()', I understand you send now more information along with the interface main inforamtion.
> now my question is: do you intend to give more information on the interface to the zclients in the same message ? I mean for instance vxlan and gre interface information should transmitted ? Or do you think some specific message should be registered ?

I have just preserved the existing behavior.

Reply:
> Interface addresses are sent in zebra_interface_address_add_update. It will be a duplicate, so I removed it in here.

I just tested your code and with this change zebra doesn't send interface addresses to the client daemons anymore (unless the addresses are configured on zebra while the other daemons are already running, in which case a different code path is executed). Please restore the call to `zsend_interface_addresses()` here and put the new call to `zsend_interface_link_params()` on a separate commit.